### PR TITLE
feat(bedrock): Temporary AWS credentials via env vars + supported models update

### PR DESCRIPTION
**Pull request by JGalego**
**Features**

- Added support for [temporary credentials](https://docs.aws.amazon.com/IAM/latest/UserGuide/id_credentials_temp.html)
- Added support for [AWS credentials via environment variables](https://docs.aws.amazon.com/cli/v1/userguide/cli-configure-envvars.html) (`AWS_*`)
   * https://github.com/geekan/MetaGPT/issues/1174#issuecomment-2096092836
- Updated supported models and corrected max tokens
   * [New models](https://docs.aws.amazon.com/bedrock/latest/userguide/model-ids.html)
      - (AI21) [Jamba-Instruct](https://aws.amazon.com/blogs/machine-learning/ai21-labs-jamba-instruct-model-is-now-available-in-amazon-bedrock/)
      - (Amazon) [Titan Text Premier](https://aws.amazon.com/about-aws/whats-new/2024/05/amazon-titan-text-premier-amazon-bedrock/)
      - (Cohere) [Command R & R+](https://aws.amazon.com/about-aws/whats-new/2024/04/cohere-command-r-r-plus-amazon-bedrock/)
      - (Mistral) [Mistral Large 2](https://aws.amazon.com/about-aws/whats-new/2024/07/mistral-large-2-foundation-model-amazon-bedrock/)
   * [Legacy/EOL](https://docs.aws.amazon.com/bedrock/latest/userguide/model-lifecycle.html)
      - (AI21) J2 Mid & Ultra for `us-west-2`
      - (Meta) Llama 2 (--> Llama 3/3.1) 

**Example**

1. Add MetaGPT config

```yaml
llm:
  api_type: 'bedrock'
  model: "amazon.titan-text-premier-v1:0"
  # 🔔 No need to add the access_key or secret_key!
```

2. Configure AWS Credentials

```bash
export AWS_ACCESS_KEY_ID=...
export AWS_SECRET_ACCESS_KEY=...
export AWS_SESSION_TOKEN=...
export AWS_DEFAULT_REGION=us-east-1
```

3. Call MetaGPT

```bash
metagpt "Create a 2048 game"
```

### CODE REVIEW

**Comment by JGalego** [comment]
@usamimeri @Wei-Jianan as authors of [Bedrock-related PRs](https://github.com/geekan/MetaGPT/pulls?q=is%3Apr+bedrock), would you mind reviewing these changes? cc: @geekan

File: metagpt/provider/bedrock/utils.py [hunk 1]
```diff
@@ -1,52 +1,84 @@
...
+    "anthropic.claude-v2:0:18k": 4096,
+    "anthropic.claude-v2:0:100k": 4096,
+    # Claude v2.1
+    "anthropic.claude-v2:1": 4096,
+    "anthropic.claude-v2:1:18k": 4096,
+    "anthropic.claude-v2:1:200k": 4096,
+    # Claude 3 Sonnet
+    "anthropic.claude-3-sonnet-20240229-v1:0": 4096,
+    "anthropic.claude-3-sonnet-20240229-v1:0:28k": 4096,
+    "anthropic.claude-3-sonnet-20240229-v1:0:200k": 4096,
```
Comment (better629) [inline metagpt/provider/bedrock/utils.py:40]: is it(4096) rights? maybe you should have a check with other models.

File: metagpt/provider/bedrock_api.py [hunk 2]
```diff
@@ -24,18 +25,19 @@ def __init__(self, config: LLMConfig):
...
         """initialize boto3 client"""
         # access key and secret key from https://us-east-1.console.aws.amazon.com/iam
-        self.__credentital_kwargs = {
-            "aws_secret_access_key": self.config.secret_key,
-            "aws_access_key_id": self.config.access_key,
-            "region_name": self.config.region_name,
+        self.__credential_kwargs = {
+            "aws_secret_access_key": os.environ.get("AWS_SECRET_ACCESS_KEY", self.config.secret_key),
+            "aws_access_key_id": os.environ.get("AWS_ACCESS_KEY_ID", self.config.access_key),
+            "aws_session_token": os.environ.get("AWS_SESSION_TOKEN", self.config.session_token),
```
Comment (better629) [inline metagpt/provider/bedrock_api.py:37]: which scenario requires this session_token? It seems that access_key/key_id is enough?

File: metagpt/provider/bedrock/bedrock_provider.py [hunk 3]
```diff
@@ -57,15 +57,52 @@ def get_choice_text_from_stream(self, event) -> str:
...
+    # (Command R/R+) https://docs.aws.amazon.com/bedrock/latest/userguide/model-parameters-cohere-command-r-plus.html
+
+    def __init__(self, model_name: str) -> None:
+        self.model_name = model_name
 
     def _get_completion_from_dict(self, rsp_dict: dict) -> str:
         return rsp_dict["generations"][0]["text"]
 
+    def messages_to_prompt(self, messages: list[dict]) -> str:
+        if "command-r" in self.model_name:
```
Comment (better629) [inline metagpt/provider/bedrock/bedrock_provider.py:71]: Due to a account limit, can you put some passed logs under using commnd-r/jamba models.

File: metagpt/provider/bedrock/utils.py [hunk 4]
```diff
@@ -1,52 +1,84 @@
...
+    # Command R+
+    "cohere.command-r-plus-v1:0": 4096,
+    # Llama 2 (--> Llama 3/3.1) !!!
+    # + Legacy: 2024-05-12
+    # + EOL: 2024-10-30
+    "meta.llama2-13b-chat-v1": 2048,
+    "meta.llama2-13b-chat-v1:0:4k": 2048,
+    "meta.llama2-70b-v1": 2048,
+    "meta.llama2-70b-v1:0:4k": 2048,
+    "meta.llama2-70b-chat-v1": 2048,
```
Comment (better629) [inline metagpt/provider/bedrock/utils.py:66]: Due to larger model can produce better result, maybe we don't need all these models, maybe keep some MoE models or size large than 30B?

File: metagpt/provider/bedrock/utils.py (hunk 1 above)
Comment (JGalego) [reply metagpt/provider/bedrock/utils.py:42]: `max_tokens`_to_sample is 4096 for every model, except for Claude 3.5 Sonnet (8192)

**References:**
* (AWS) [Anthropic Claude Text Completions API > Request and Response](https://docs.aws.amazon.com/bedrock/latest/userguide/model-parameters-anthropic-claude-text-completion.html#model-parameters-anthropic-claude-text-completion-request-response)
* (Anthropic) [Model comparison](https://docs.anthropic.com/en/docs/about-claude/models#model-comparison)

File: metagpt/provider/bedrock/bedrock_provider.py (hunk 3 above)
Comment (JGalego) [reply metagpt/provider/bedrock/bedrock_provider.py:71]: Can you elaborate on this one?

File: metagpt/provider/bedrock/utils.py (hunk 4 above)
Comment (JGalego) [reply metagpt/provider/bedrock/utils.py:71]: I'll comment out the legacy Llama 2 models (going EOL at EOM cf. [Model Lifecycle > Legacy Versions](https://docs.aws.amazon.com/bedrock/latest/userguide/model-lifecycle.html#versions-for-eol) for more information), the smaller (`<30B`) Llama 3 models and Mistral 7B Instruct. **Anything else?**

File: metagpt/provider/bedrock/bedrock_provider.py (hunk 3 above)
Comment (better629) [reply metagpt/provider/bedrock/bedrock_provider.py:71]: The log of running commnd-r/jamba models like executing `examples/hello_world.py` through asking `who are you`.

Comment (JGalego) [reply metagpt/provider/bedrock/bedrock_provider.py:71]: I don't understand what you're getting at and looking at `exmaples/hello_world.py` didn't help. The other `message_to_prompt` implementations are not logging anything anyway. Feel free to edit this PR directly if it helps.

**Comment by better629** [comment]
lgtm

File: metagpt/provider/bedrock/bedrock_provider.py (hunk 3 above)
Comment (better629) [reply metagpt/provider/bedrock/bedrock_provider.py:71]: I don't have a aws account to test command-r/jamba models. So in the previous comment, I wanted you to post some test results(directly using `examples/hello_world.py`) to show that it works.

**Review by geekan — APPROVED** [review]
LGTM

**Pull request merged**